### PR TITLE
allow suffixes to be set in locale

### LINKED ISSUE #2: enable locale support to formatAbbreviate()
Currently `formatAbbreviate()` only format the numbers using `en-US` standard. The idea is enabling a new property that allows to set custom format to the numbers according the `locale` used in the site.

### PR DESCRIPTION
**Pull request by cnavarreteliz**
(closes #2)

This PR allows customize the suffixes of the numbers that will be set using the locale. It includes support for +10 locales, and the user will can set its custom suffixes.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have documented all new methods.
- [x] I have written tests for all new methods/functionality.
- [ ] I have written examples for all new methods/functionality.

### CODE REVIEW

File: src/abbreviate.js [hunk 1]
```diff
@@ -1,25 +1,59 @@
...
-    @param {Number} n The number to be formatted.
+    @param {Number|String} n The number to be formatted.
+    @param {Object|String} locale The locale config to be used. If *value* is an object, the function will format the numbers according the object. The object must include `suffixes`, `delimiter` and `currency` properties.
     @returns {String}
 */
-export default function(n) {
-  if (typeof n !== "number") return "N/A";
-  const length = n.toString().split(".")[0].replace("-", "").length;
+export default function(n, locale = "en-US") {
+  if (isFinite(n) && !isNaN(n)) n *= 1;
```
Comment (davelandry) [inline src/abbreviate.js]: oooo I never knew about `ifFinite`, I like it!

It seems to also catch for `NaN`, so you can probably remove the `isNaN` catch here

File: src/locale.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    "suffixes": ["y", "z", "a", "f", "p", "n", "µ", "m", "", "k", "M", "B", "t", "q", "Q", "Z", "Y"],
+    "delimiters": {
+      "thousands": ".",
+      "decimal": ","
+    },
+    "currency": {
+      "symbol": "$"
+    }
+  },
+  "et": {
```
Comment (davelandry) [inline src/locale.json]: this should be `"et-EE"`

File: src/locale.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    "suffixes": ["y", "z", "a", "f", "p", "n", "µ", "m", "", "tuh", "mln", "mld", "trl", "q", "Q", "Z", "Y"],
+    "delimiters": {
+      "thousands": " ",
+      "decimal": ","
+    },
+    "currency": {
+      "symbol": "€"
+    }
+  },
+  "fr": {
```
Comment (davelandry) [inline src/locale.json]: this should be `"fr-FR"`

File: src/locale.json [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    "suffixes": ["y", "z", "a", "f", "p", "n", "µ", "m", "", "k", "mm", "b", "t", "q", "Q", "Z", "Y"],
+    "delimiters": {
+      "thousands": ".",
+      "decimal": ","
+    },
+    "currency": {
+      "symbol": "€"
+    }
+  },
+  "es-LA": {
```
Comment (davelandry) [inline src/locale.json]: Is this one for representing **L**atin **A**merica? I only want to use language-country codes that are part of the standard ISO set. So if you want one for Chile, can we do `"es-CL"`? I found this list of codes online: https://www.fincher.org/Utilities/CountryLanguageList.shtml

File: src/abbreviate.js [hunk 5]
```diff
@@ -1,25 +1,59 @@
...
   else if (length >= 3) {
-    const f = format(".3s")(n)
-      .replace("G", "B")
-      .replace("T", "t")
-      .replace("P", "q")
-      .replace("E", "Q");
-    const num = f.slice(0, -1);
-    const char = f.slice(f.length - 1);
-    val = `${parseFloat(num)}${char}`;
+    const f = formatSuffix(n, 2, suffixes);
```
Comment (davelandry) [inline src/abbreviate.js]: Is there a reason you're not using d3-format here? I can understand not wanting to use the `".3s"` scientific formatting, as then you're just swapping out the scientific number, but why not just use `".3r"`, and then just determine the suffix yourself?

File: src/abbreviate.js [hunk 6]
```diff
@@ -1,29 +1,70 @@
...
   else if (length >= 3) {
-    const f = format(".3s")(n)
-      .replace("G", "B")
-      .replace("T", "t")
-      .replace("P", "q")
-      .replace("E", "Q");
-    const num = f.slice(0, -1);
-    const char = f.slice(f.length - 1);
-    val = `${parseFloat(num)}${char}`;
+    const f = formatSuffix(d3plusFormatLocale.format(".3r")(n), 2, suffixes);
```
Comment (davelandry) [inline src/abbreviate.js]: Now you're using BOTH the d3 number formatter and your own! Shouldn't the final logic be something more like this?

```js
val = `${d3plusFormatLocale.format(".3r")(n)}${separator}${char}`;
```

File: test/abbreviate.js [hunk 7]
```diff
@@ -20,6 +20,9 @@ test("abbreviate", assert => {
   assert.equal("1B",    abbreviate(1000000009), "large - removes trailing zeros and period");
   assert.equal("0.1",   abbreviate(0.1), "small - removes trailing zeros");
 
+  assert.equal("10 mln", abbreviate(10000000, "et-EE"), "estonian locale");
+  assert.equal("1mm",    abbreviate(1000000, "es-ES"), "spanish locale");
+
```
Comment (davelandry) [inline test/abbreviate.js]: This is great. Can you also add a few tests for the locale delimiters?

File: src/abbreviate.js (hunk 6 above)
Comment (cnavarreteliz) [reply src/abbreviate.js]: @davelandry `d3plusFormatLocale.format(".3r")(n)` only returns the number in decimal notation, rounded to significant digits.

```
d3plusFormatLocale.format(".3r")(123456) -> 123000
```

and `formatSuffix` returns the suffix and the first characters of the number

Comment (davelandry) [reply src/abbreviate.js]: Ahh ok interesting, I guess I didn't understand how the format locale works (the other d3 format functions usually return an abbreviated string). Get a few more test in there for delimiters and we should be good to go!

**Pull request merged**